src/main/java/seedu/duke/foodreferencelists/StallsManager.java

```diff
@@ -116,9 +129,39 @@ public static void printAllItems() {
         for (int storeIndex : idName.keySet()) {
             String storeName = idName.get(storeIndex);
             System.out.println(storeName);
-            Ui.printLine();
             printItems(storeIndex);
         }
+        Ui.printMessage("Wow, thats a lot of options! Finished printing");
+        Ui.printLine();
+    }
+
+    public static void printFoodRecordsWithLowerCalories(int calories) {
+        ArrayList<WhatIAteList> allLists = new ArrayList<>();
+        assert MAX_STORE_INDEX > 0; //checks that data dump is initialised
+        for (int storeIndex : idName.keySet()) {
+            WhatIAteList storeItems = parseStringIntoFoodRecordList(storeIndex);
+            allLists.add(storeItems);
+        }
+        WhatIAteList filterLowerThanCalories = new WhatIAteList();
+        for (WhatIAteList storeItems : allLists) {
+            for (FoodRecord foodItem : storeItems.getList()) {
+                if (foodItem.getCalorieCount() <= calories) {
+                    filterLowerThanCalories.addToList(foodItem, true);
+                }
+            }
+        }
+        //sort by calorie count
+        filterLowerThanCalories.getList().sort(
+            (FoodRecord f1, FoodRecord f2)
+                -> f1.getCalorieCount().compareTo(f2.getCalorieCount()));
+        //sort by name
+        filterLowerThanCalories.getList().sort(
+            (FoodRecord f1, FoodRecord f2)
+                -> f1.getFoodName().compareTo(f2.getFoodName()));
```

Comment:
You can extract a method to sort the callories. Btw good use of streams!

Reply:
Nice catch, just realized I can  extract the method rather than adding a `//comment` to explain what I'm doing there haha